src/orion/storage/base.py

```diff
@@ -558,27 +558,17 @@ def acquire_algorithm_lock(self, experiment, timeout=600, retry_interval=1):
         raise NotImplementedError()
 
 
-storage_factory = GenericSingletonFactory(BaseStorageProtocol)
+storage_factory = GenericFactory(BaseStorageProtocol)
 
 
-def get_storage():
-    """Return current storage
+@contextlib.contextmanager
+def with_storage(config):
```

Comment:
Is this used for tests only?

Reply:
I dont think it is used right now, it was a helper, could become useful. want it removed ?

Reply:
it is removed

Reply:
Yeah, less maintenance. Thanks!